src/main/java/io/odpf/firehose/objectstorage/ObjectStorageFactory.java

```diff
@@ -0,0 +1,21 @@
+package io.odpf.firehose.objectstorage;
+
+import io.odpf.firehose.objectstorage.gcs.GCSConfig;
+import io.odpf.firehose.objectstorage.gcs.GCSObjectStorage;
+
+import java.io.IOException;
+
+public class ObjectStorageFactory {
+
+    public static ObjectStorage createObjectStorage(ObjectStorageType storageType, Object config) {
+        if (storageType == ObjectStorageType.GCS) {
+            try {
+                GCSConfig gcsConfig = (GCSConfig) config;
```

Comment:
ah. can we find a better way other than casting?

Reply:
this is tricky, this used on `DLQWriterFactory`, as well as` ObjectStorageSinkFactory`, so we have to make the parameter generic, because in the future we might have, `AWSS3Config` or `HDFSConfig`, 
i'm thinking of key values map, will try to find another approach before discuss this

Reply:
Yes. Casting here is not good!

Reply:
changed the implementation by passing config using `Map<String,Object>` argument